### DISCUSSION TITLE
Verbindung Testen , Test Connection

### DISCUSSION THREAD

**Discussion by Clluberlang** [wimaha/TeslaBleHttpProxy#46] [General]
Ist es möglich die Verbindung zum Tesla zu testen, oder EVCC zu testen benutzen und gucken ob die Verbindung genutzt wird. 

Is it possible to test the connection to the Tesla ?

**Comment by D4rkH0ur** [comment]
Du kannst im Log sehen ob die Verbindung funktioniert. Man sieht ob die Verbindung geklappt hat oder nicht. Und welche Kommandos gesendet werden und ob diese angenommen oder abgelehnt wurden.

**Reply by Clluberlang** [reply]
Ok danke das habe ich schon mitbekommen , hatte gestern nämlich die ganze Zeit diese Meldung und die Ladestärke wurde nicht angepasst.

[lp-1  ] ERROR 2024/08/16 13:42:42 charger disable: Post "http://192.168.178.149:8080/api/1/vehicles/XP7XXXXXXXXXXXX/command/charge_stop": net/http: request canceled (Client.Timeout exceeded while awaiting headers)

Heute habe ich mal das „http://„ aus der Proxy Adresse raus genommen obwohl bei Einrichtung hat es ja am Anfang kurz funktioniert. Zudem habe ich einen Neustart von EVCC gemacht, jetzt gerade wurde das laden kurz geregelt und angehalten. Leide weitere Tests im Moment nicht möglich, bewölkt und Regen 🌧️. Mal sehen

**Reply by D4rkH0ur** [reply]
Das ist der Log von EVCC oder ? 
Ich meine den vom Proxy. Da siehst du was passiert oder auch nicht. 

Dein Fehler schaut aus wie das aktuelle API Limit von Tesla zu sein. Weshalb es aktuell nur mit den CommandProxy funktioniert.

**Reply by Clluberlang** [reply]
Ja ist das log vom EVCC , das API Limit kann es eigentlich nicht sein als heute Morgen mein WC freigeschaltet wurde wurde munter drauf los geladen. Erst nach dem Neustart vom EVCC wurde die Ladesteuerung wieder aktiv.

**Reply by D4rkH0ur** [reply]
Die Steuerung über die Tesla API funktioniert auch jeden Tag immer wieder. Nur eben sehr sehr kurz. Dann kommt deine Fehlermeldung weil das Limit erreicht wurde. 

Einfach in den Log vom TeslaBleHttpProxy schauen. Da kannst du sehen ob er was macht.

**Reply by Clluberlang** [reply]
Sorry bin Anfänger was Raspi und ssh angeht, gucke gerade wie ich ans logfile vom Proxy komme. Aber wie gesagt das Limit kann es ja nicht sein da die Steuerung heute direkt nicht ging . Hatte ja vorher die Steuerung über Tesla API und da hat es ja funktioniert bis das Limit erreicht wurde. Aber mit dem Proxy ging es gar nicht heute Morgen. Erst nach dem Neustart vom EVCC.

**Reply by Clluberlang** [reply]
Ok habe das logfile vom Proxy . Also keine Fehler wie ich das sehe , oder sehe ich das falsch? 
Muss also der Fehr vom EVCC kommen. 
Er hätte heute Morgen nicht laden dürfen , mein Hausakku hatte sein Limit noch nicht erreicht ! 


2024/08/17 11:01:04 INFO received command=set_charging_amps body=map[charging_amps:7]
2024/08/17 11:01:33 INFO received command=set_charging_amps body=map[charging_amps:7]
2024/08/17 11:02:03 INFO received command=set_charging_amps body=map[charging_amps:7]
2024/08/17 11:02:30 INFO received command=set_charging_amps body=map[charging_amps:6]
2024/08/17 11:03:01 INFO received command=set_charging_amps body=map[charging_amps:7]
2024/08/17 11:03:32 INFO received command=set_charging_amps body=map[charging_amps:6]
2024/08/17 11:04:03 INFO received command=set_charging_amps body=map[charging_amps:7]
2024/08/17 11:04:35 INFO received command=set_charging_amps body=map[charging_amps:7]
2024/08/17 11:05:10 INFO received command=set_charging_amps body=map[charging_amps:7]
2024/08/17 11:05:27 INFO received command=set_charging_amps body=map[charging_amps:7]
2024/08/17 11:06:06 INFO received command=set_charging_amps body=map[charging_amps:7]
2024/08/17 11:06:30 INFO received command=set_charging_amps body=map[charging_amps:7]
2024/08/17 11:07:01 INFO received command=set_charging_amps body=map[charging_amps:7]
2024/08/17 11:07:29 INFO received command=set_charging_amps body=map[charging_amps:7]
2024/08/17 11:08:01 INFO received command=set_charging_amps body=map[charging_amps:7]
2024/08/17 11:08:33 INFO received command=set_charging_amps body=map[charging_amps:5]
2024/08/17 11:09:02 INFO received command=set_charging_amps body=map[charging_amps:7]
2024/08/17 11:09:35 INFO received command=set_charging_amps body=map[charging_amps:7]
2024/08/17 11:10:01 INFO received command=set_charging_amps body=map[charging_amps:7]
2024/08/17 11:10:32 INFO received command=set_charging_amps body=map[charging_amps:7]
2024/08/17 11:11:02 INFO received command=set_charging_amps body=map[charging_amps:7]
2024/08/17 11:11:30 INFO received command=set_charging_amps body=map[charging_amps:7]
2024/08/17 11:12:01 INFO received command=set_charging_amps body=map[charging_amps:7]
2024/08/17 11:12:34 INFO received command=set_charging_amps body=map[charging_amps:6]
2024/08/17 11:13:04 INFO received command=set_charging_amps body=map[charging_amps:5]
2024/08/17 11:13:38 INFO received command=set_charging_amps body=map[charging_amps:7]
2024/08/17 11:14:00 INFO received command=set_charging_amps body=map[charging_amps:7]
2024/08/17 11:14:36 INFO received command=set_charging_amps body=map[charging_amps:7]
2024/08/17 11:15:04 INFO received command=set_charging_amps body=map[charging_amps:7]
2024/08/17 11:15:32 INFO received command=set_charging_amps body=map[charging_amps:7]
2024/08/17 11:16:01 INFO received command=set_charging_amps body=map[charging_amps:7]
2024/08/17 11:16:32 INFO received command=set_charging_amps body=map[charging_amps:5]
2024/08/17 11:17:03 INFO received command=set_charging_amps body=map[charging_amps:5]
2024/08/17 11:17:31 INFO received command=set_charging_amps body=map[charging_amps:5]
2024/08/17 11:18:03 INFO received command=set_charging_amps body=map[charging_amps:5]
2024/08/17 11:18:31 INFO received command=set_charging_amps body=map[charging_amps:5]
2024/08/17 11:19:01 INFO received command=set_charging_amps body=map[charging_amps:5]
2024/08/17 11:19:31 INFO received command=set_charging_amps body=map[charging_amps:7]
2024/08/17 11:20:00 INFO received command=set_charging_amps body=map[charging_amps:5]
2024/08/17 11:20:34 INFO received command=set_charging_amps body=map[charging_amps:5]
2024/08/17 11:21:06 INFO received command=set_charging_amps body=map[charging_amps:5]

**Reply by D4rkH0ur** [reply]
Ich sehe keine Verbindung zum Auto. Nur das dein Proxy Kommandos empfangen hat von EVCC. Schau mal im Dashboard ob du erfolgreich die Keys mit dem Auto verbunden hast.

**Reply by Clluberlang** [reply]
Also eingentlich hat es geklappt . Im Tesla ist der Schlüüsel auch aufgeführt. 
das Dashboard sieht so aus !
![IMG_0661](https://github.com/user-attachments/assets/fdc12723-881b-4adf-bfa5-9bfbc2f05db4)

**Reply by wimaha** [reply]
Bitte mal den Proxy neustarten.

**Reply by TobiEdelberg** [reply]
Ich habe das gleiche Problem. 

Zuerst sieht es so aus:
2024/08/17 13:02:57 INFO connecting to Vehicle ...
2024/08/17 13:03:01 INFO connection established
2024/08/17 13:03:01 INFO sending command=set_charging_amps body=map[charging_amps:7]
2024/08/17 13:03:03 INFO successfully executed command=set_charging_amps body=map[charging_amps:7]
2024/08/17 13:03:32 INFO received command=set_charging_amps body=map[charging_amps:5]
2024/08/17 13:03:33 INFO connecting to Vehicle ...
2024/08/17 13:03:37 INFO connection established
2024/08/17 13:03:37 INFO sending command=set_charging_amps body=map[charging_amps:5]
2024/08/17 13:03:38 INFO successfully executed command=set_charging_amps body=map[charging_amps:5]
2024/08/17 13:06:39 INFO received command=set_charging_amps body=map[charging_amps:6]
2024/08/17 13:06:39 INFO connecting to Vehicle ...
2024/08/17 13:06:55 WARN failed to connect to vehicle (C): context deadline exceeded
2024/08/17 13:06:55 INFO retrying in 3 seconds
2024/08/17 13:07:03 INFO connection established

Und ab einem bestimmten Zeitpunkt geht dann nix mehr ohne eine Änderung... Da bleibt es hängen mit connecting to vehicle und dann sieht man nur noch die eingehenden Befehle von EVCC. 

2024/08/17 13:06:55 WARN failed to connect to vehicle (C): context deadline exceeded
2024/08/17 13:06:55 INFO retrying in 3 seconds
2024/08/17 13:07:03 INFO connection established
2024/08/17 13:07:03 INFO sending command=set_charging_amps body=map[charging_amps:6]
2024/08/17 13:07:04 INFO successfully executed command=set_charging_amps body=map[charging_amps:6]
2024/08/17 13:07:36 INFO received command=set_charging_amps body=map[charging_amps:5]
2024/08/17 13:07:36 INFO connecting to Vehicle ...
2024/08/17 13:08:01 INFO received command=set_charging_amps body=map[charging_amps:5]
2024/08/17 13:08:26 INFO received command=set_charging_amps body=map[charging_amps:5]
2024/08/17 13:08:56 INFO received command=set_charging_amps body=map[charging_amps:5]
2024/08/17 13:09:26 INFO received command=set_charging_amps body=map[charging_amps:5]
2024/08/17 13:09:56 INFO received command=set_charging_amps body=map[charging_amps:4]
2024/08/17 13:10:26 INFO received command=set_charging_amps body=map[charging_amps:2]
2024/08/17 13:10:56 INFO received command=set_charging_amps body=map[charging_amps:2]

**Reply by Clluberlang** [reply]
@wimaha Danke erstmal , habe Proxy HA und EVCC mal neu gestartet. Werde es jetzt mal beobachten . Melde mich mit Updates.

**Comment by Clluberlang** [comment]
@wimaha also das Problem besteht weiterhin , heute hätte das Fahrzeug gegen 12:37h anfangen sollen zu laden. Jedoch konnte keine Verbindung zum Fahrzeug hergestellt werden. habe dann um 14:02 h den Proxy neu gestartet hat leider immer noch nicht funktioniert.
Um 14:06h habe ich dann EVCC neu gestartet und dann hat es funktioniert. Finde im Moment den Fehler nicht .
PS Der pi mit dem Proxyserver wurde in der ganzen zeit nicht bewegt.
Logfile vom TeslaBleHttpProxy und die Config vom EVCC hänge ich mal an .

Hoffe jemand kann mir helfen.

[logfile_TeslaBleHttpProxy.log](https://github.com/user-attachments/files/16660195/logfile_TeslaBleHttpProxy.log)

[evcc.TXT](https://github.com/user-attachments/files/16660238/evcc.TXT)

**Reply by D4rkH0ur** [reply]
Vielleicht die zwei Keys in EVCC erneuern. accessToken und refreshToken

**Reply by Clluberlang** [reply]
@D4rkH0ur Werde ich versuchen. 
jetzt gerade auch wieder ein Halbe Stunde hat er gesteuert und dann war wieder vorbei.


2024/08/19 14:43:12 INFO received command=set_charging_amps body=map[charging_amps:4]
2024/08/19 14:43:12 INFO connecting to Vehicle ...
2024/08/19 14:43:15 INFO connection established
2024/08/19 14:43:15 INFO sending command=set_charging_amps body=map[charging_amps:4]
2024/08/19 14:43:16 INFO successfully executed command=set_charging_amps body=map[charging_amps:4]
2024/08/19 14:43:41 INFO received command=set_charging_amps body=map[charging_amps:6]
2024/08/19 14:43:41 INFO sending command=set_charging_amps body=map[charging_amps:6]
2024/08/19 14:43:42 INFO successfully executed command=set_charging_amps body=map[charging_amps:6]
2024/08/19 14:44:42 INFO received command=set_charging_amps body=map[charging_amps:4]
2024/08/19 14:44:42 INFO connecting to Vehicle ...
2024/08/19 14:44:45 INFO connection established
2024/08/19 14:44:45 INFO sending command=set_charging_amps body=map[charging_amps:4]
2024/08/19 14:44:46 INFO successfully executed command=set_charging_amps body=map[charging_amps:4]
2024/08/19 14:45:16 INFO received command=set_charging_amps body=map[charging_amps:5]
2024/08/19 14:45:16 INFO connecting to Vehicle ...
2024/08/19 14:45:44 INFO received command=set_charging_amps body=map[charging_amps:7]
2024/08/19 14:46:16 INFO received command=set_charging_amps body=map[charging_amps:16]
2024/08/19 14:46:43 INFO received command=set_charging_amps body=map[charging_amps:14]
2024/08/19 14:47:15 INFO received command=set_charging_amps body=map[charging_amps:9]
2024/08/19 14:47:45 INFO received command=set_charging_amps body=map[charging_amps:13]
2024/08/19 14:48:18 INFO received command=set_charging_amps body=map[charging_amps:6]
2024/08/19 14:49:20 INFO received command=set_charging_amps body=map[charging_amps:4]
2024/08/19 14:49:40 INFO received command=set_charging_amps body=map[charging_amps:3]
2024/08/19 14:50:13 INFO received command=set_charging_amps body=map[charging_amps:3]
2024/08/19 14:50:41 INFO received command=set_charging_amps body=map[charging_amps:3]
2024/08/19 14:51:10 INFO received command=set_charging_amps body=map[charging_amps:3]

**Reply by D4rkH0ur** [reply]
Ah dann besteht ja auf alle Fälle eine funktionierende Verbindung. Immerhin. Ist der Pi eventuell zu weit vom Auto entfernt ?

**Reply by Clluberlang** [reply]
keine drei Meter durch ein Fenter, finde ich dann aber sehr merkwürdig. Weil nach EVCC Neustart es dann wieder Funktioniert. hatte gerade die die Tokens erneuert und musste HA und EVCC gefühlt drei mal neu Starten bis es wieder ging. finde ich sehr merkwürdig.

**Reply by wimaha** [reply]
Kannst du bitte den Proxy mit Debug Logs laufen lassen?
Dafür in der docker-compose folgendes ergänzen:
```
environment: #optional
      logLevel: debug
```

So, wie es hier https://github.com/wimaha/TeslaBleHttpProxy/blob/main/docker-compose.yml zu sehen ist.

Das irgendwann die Befehle nicht mehr ausgeführt werden, ist ein Bug, der vermutlich im BLE-Package liegt ... den versuche ich derzeit ausfindig zu machen.

**Reply by Clluberlang** [reply]
@wimaha danke habe ich gemacht , sobald mir wieder ein Ausfall auffällt lade ich das logfile hoch .

**Reply by Clluberlang** [reply]
@wimaha Ok nun war es wieder soweit , Kommunikation abgebrochen . Zuerst Neustart Proxy dann Neustart EVCC dann geht es wieder . Nur Neustart Proxy keine Abhilfe !
Log im Anhang
[Log-TeslaBle.txt](https://github.com/user-attachments/files/16662168/Log-TeslaBle.txt)

**Reply by wimaha** [reply]
Danke, leider war der Debug Mode nicht an …
Hattest du dem aktiviert? Ggf. Musst du einmal docker compose up -d machen.

**Reply by Clluberlang** [reply]
@wimaha Hatte die Compose ergänzt und einen restart gemacht . Up -d hatte ich nicht gemacht , bin Anfänger . Werde ich jetzt noch mal machen . Gibt dann morgen wieder Ergebnisse. Danke dir schönen Abend noch .

**Reply by wimaha** [reply]
Kein Problem! 😃 nach dem Start steht es auch in den Logs, wenn Debug an ist.

**Reply by Clluberlang** [reply]
Ja habe ich gerade gesehen ! Danke noch mal.

**Reply by Clluberlang** [reply]
@wimaha grüße , habe nun das Log . Hoffe es kann bei der Lösung helfen . 😉
[log-ble.txt](https://github.com/user-attachments/files/16676684/log-ble.txt)

**Reply by wimaha** [reply]
Danke! Es gibt parallel eine Issue in der https://github.com/teslamotors/vehicle-command Library. Dort habe ich das Log gepostet. Hoffentlich finden wir den Fehler...

**Reply by wimaha** [reply]
Der Bug ist mit Version 1.2.4 behoben. 
`docker compose pull` und danach `docker compose up -d` zum Updaten. :-)

**Reply by Clluberlang** [reply]
@wimaha danke dir für deinen Einsatz , Update ist gemacht. Werde berichten wie es sich verhält!

**Reply by Clluberlang** [reply]
@wimaha kleines Zwischenfazit , bis jetzt sieht es gut aus. Keine Aussetzer gestern und heute. 👍Gute Arbeit.

**Comment by Clluberlang** [comment]
@wimaha danke erstmal für deinen Einsatz, und die Geduld bei einem noob. Nun berichte ich mal, eine Woche ist vergangen und es funktioniert alles tadellos. Verbindung ist stabil und läuft seit einer Woche ohne Probleme.
